myfitnesspal/base.py

```diff
@@ -1,6 +1,13 @@
+import six
+
 class MFPBase(object):
     def __str__(self):
-        return self.__unicode__().encode('ascii', 'replace')
+        #return self.__unicode__().encode('ascii', 'replace')
+        # return unicode for both Python 2.7 and 3.4 PORTING_CHECK
```

Comment:
Ahh -- I'm afraid you do have to return bytes on Python 2.7 :-/.  I think your below `if six.PY2`/`else` bit is exactly what you need.